### PR TITLE
support optional arguments

### PR DESCRIPTION
**Pull request by magnurud**

### CODE REVIEW

File: las/client.py [hunk 1]
```diff
@@ -178,10 +177,7 @@ def get_asset(self, asset_id: str) -> Dict:
     def update_asset(
         self,
         asset_id: str,
-        *,
-        content: Optional[bytes] = None,
-        name: Optional[str] = None,
-        description: Optional[str] = None,
+        **optional_args,
```
Comment (augustak) [inline las/client.py:180]: Can this fit on one line now?

File: las/client.py [hunk 2]
```diff
@@ -203,14 +199,12 @@ def update_asset(
...
  :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
         """
-        body = {
-            'content': b64encode(content).decode() if content is not None else None,
-            'name': name,
-            'description': description,
-        }
-        return self._make_request(requests.patch, f'/assets/{asset_id}', body=dictstrip(body))
+        if content := optional_args.get('content'):
+            optional_args['content'] = b64encode(content).decode()
```
Comment (augustak) [inline las/client.py:203]: Is there a point checking the type of content? If bytes then b64encode if str then assume it's a filepath if Path then assume it's a file path?

Comment (augustak) [reply las/client.py:203]: Or if str assume it's already b64encoded?

File: las/client.py [hunk 3]
```diff
@@ -480,8 +492,7 @@ def update_secret(
         secret_id: str,
         *,
         data: Optional[dict] = None,
-        name: Optional[str] = None,
-        description: Optional[str] = None,
+        **optional_args,
```
Comment (augustak) [inline las/client.py:495]: Can this fit on one line now?

File: las/client.py [hunk 4]
```diff
@@ -821,9 +833,8 @@ def create_workflow(
         self,
         specification: dict,
         *,
-        name: Optional[str] = None,
-        description: Optional[str] = None,
         error_config: Optional[dict] = None,
+        **optional_args,
```
Comment (augustak) [inline las/client.py:837]: Can this fit on one line now?

File: las/client.py [hunk 5]
```diff
@@ -901,11 +912,8 @@ def update_workflow(self, workflow_id: str, *, name: Optional[str], description:
         :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
  :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
         """
-        body = {
-            'name': name,
-            'description': description,
-        }
-        return self._make_request(requests.patch, f'/workflows/{workflow_id}', body=dictstrip(body))
+        body = optional_args
```
Comment (augustak) [inline las/client.py:915]: body is not needed

File: tests/test_assets.py [hunk 6]
```diff
@@ -38,9 +43,14 @@ def test_get_asset(client: Client):
     assert 'content' in response, 'Missing content in response'
 
 
-def test_update_asset(client: Client):
+@pytest.mark.parametrize('name,description', [
+    ('foo', ''),
+    ('', 'foo'),
+    ('foo', 'bar'),
```
Comment (augustak) [inline tests/test_assets.py:49]: is it an idea to test all cases? explicit None, str and not provided?

Comment (magnurud) [reply tests/test_assets.py:49]: None fails for unknown reasons... but could add not provided

**Review by augustak — APPROVED** [review]

**Pull request merged**